SOP node helpdesk ticket classification (#49)
GenomicDataInfrastructure/standard-operating-procedures: sops/node-specific/GDI-SOP0008_sop-node-helpdesk-ticket-classification.md

> # European GDI - SOP Node Helpdesk Ticket Classification

Reviewer: Is the title expected to be "*SOP* Node Helpdesk..."? It may be that *SOP* at the beginning, both in the filename and in this title, was a trail from the SOP0007 that had that in the filename and title (?)

In other words, I would rename the file and the title to ``Node Helpdesk Ticket Classification`` (and filename in lowercase)

Author: Ok, I did it!

4. [Purpose](#4-purpose)

5. [Scope](#5-scope)

6. [Introduction and Background Information](#6-introduction-and-background-information)

7. [Summary or Context Diagram](#7-summary-or-context-diagram)

8. [Procedure](#8-procedure)

9. [References](#9-references)

### 1. Document History

| Template Version | Instance version | Author(s) | Description of changes | Date |
|---|---|---|---|---|
|  ``v1``  |  | Elisavet Torstensson, Mattias Strömberg, Silvia Bahena| Created the first version of the SOP | 2025.02.04 |

### 2. Glossary

Find GDI SOPs common Glossary at the [charter document](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_charter.md).

| Abbreviation |  Description |
|---|---|
| EBI | European Bioinformatics Institute |
| EMBL | European Molecular Biology Laboratory |
| GDI | Genomic Data Infrastructure |
| ISM | Information Service Management |
| NBIS | National Bioinformatics Infrastructure Sweden |
| NHD | Node Helpdesk |
| ORR | Organisational Roles and Responsibilities |
| SOP | Standard Operating Procedure |
| UU | University of Uppsala |
| VHD | Virtual Helpdesk  |

| Term | Definition |
|---|---|

> | Node Helpdesk | The support team is responsible for handling tickets related to data management issues associated with the node in the GDI Project.  |

Reviewer: ```suggestion
| Node Helpdesk | The support team responsible for handling tickets related to data management issues associated with the node in the GDI Project. |
```

> | Ticket | Incoming issues from the GDI Virtual Helpdesk, are to be processed within the Node Helpdesk and, after that, returned to the Virtual Helpdesk |

Reviewer: ```suggestion
| Ticket | Incoming issues from the GDI Virtual Helpdesk that are to be processed within the Node Helpdesk and, after that, returned to the Virtual Helpdesk |
```

> | Ticket Classification | The process of categorising support requests is based on their nature so that they can be assigned to the appropriate individual or team for resolution.  |

Reviewer: ```suggestion
| Ticket Classification | The process of categorising support requests based on their nature, so they can be assigned to the appropriate individual or team for resolution.  |
```

> | Virtual Helpdesk  | The initial point of contact for inquiries and requests related to data management in the GDI project. |

Reviewer: Are HelpDesks in GDI and nodes only to deal with data management? Just a curiosity, since I'm used to HD teams dealing with broader topics

Reviewer 2: I think saying "data management" covers in a general way the HD tasks i.e submission, administration, curation, distribution

See the qualifications and responsibilities of the roles in the [Organisational Roles and Responsibilities](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_organisational-roles-and-responsibilities.md) document.

| Role | Full name | GDI/node role | Organisation |
|---|---|---|---|
| Author | Elisavet Torstensson | Task 4.3 member  | UU-NBIS |
| Author | Mattias Strömberg | Task 4.3 member  | UU-NBIS |
| Author | Silvia Bahena | Task 4.3 member  | EMBL-EBI |
| Reviewer |  |  |  |
| Approver |  |  |  |
| Authorizer |  |  |  |

### 4. Purpose

Each GDI Node Helpdesk (NHD) provides support to its users by using a dedicated system (ticket management system) to respond to users with various issues or requests.

The purpose of this Standard Operating Procedure (SOP) is to provide clear guidelines for classifying tickets in a consistent and efficient manner in the node’s dedicated system. Proper ticket classification ensures that requests are routed to the appropriate teams, and prioritised accurately. This SOP aims to optimize resource allocation, improve response time and enhance overall customer satisfaction.

### 5. Scope

> This SOP outlines the process for classifying tickets at the **node level** received from the Virtual Helpdesk (VHD) using a classification model and registering them in the node's internal ticket management system. The SOP covers all steps, from the reference of the ticket from the VHD to its categorisation of a corresponding issue in the dedicated system.

Reviewer: ```suggestion
This SOP outlines the process for classifying tickets, at the **node level**, that were received from the Virtual Helpdesk (VHD). The approach is to use a classification model and register the tickets in the node's internal ticket management system. This SOP covers the steps between the initial reference of the ticket from the VHD, to its categorisation as a corresponding issue in the dedicated system.
```

- **Enhanced Communication and Coordination:** Tags clarify ticket statuses and denote special handling requirements, improving team communication.

- **Streamlined Workflows:** Tags support automated workflows and ensure consistent ticket management practices.

- **Tracking and Reporting**: Tags aid in tracking performance metrics and analysing ticket trends for process improvements.	

- **Increased Efficiency:** Tags facilitate quick access to tickets and enable focused resolution of critical issues.

### 7. Summary or Context Diagram

This is a proposed workflow for how the ticket will be classified at the Node level.

````mermaid
flowchart TD
A[Receive Ticket from VHD] --> B{Evaluate Ticket Scope}
B -->|Positive Evaluation| C[Create New Issue in Node Ticket Management System]
C --> D[Assign Tags to Ticket
Tags: 1.Type 2.Priority 3.Assignee 4.Status]
D --> E[Follow SOP for NHD Ticket Handling]
B -->|Negative Evaluation| G[Create New Issue in Node Ticket Management System for Documentation]
G --> H[Inform VHD of Out-of-Scope Decision, Provide Explanation and Close Issue with Notes]

> ````

Reviewer: ```suggestion
````mermaid
flowchart TD
    subgraph Virtual_Helpdesk
        A[Ticket]
    end

    subgraph Node_Helpdesk
        A -->|Receive Ticket from VHD| B{Evaluate
                                        Ticket Scope}
        subgraph Ticket Management System
            C[Create Issue in Node Ticket Management System]
            F[Create 'Documentation' Issue in Node Ticket Management System]
        end                                        
        B -->|Positive Evaluation| C
        C --> D[Assign Tags: Type, Priority, Assignee, Status]
        D --> E[Follow NHD SOP for Ticket Handling]

        B -->|Negative Evaluation| F
        F --> D
    end

    E -->|Update VHD on Ticket Resolution| Virtual_Helpdesk
    F -->|Inform VHD of Out-of-Scope Decision| Virtual_Helpdesk
````
```

Reviewer: Check the diagram I drafted from yours. I think it's slightly clearer on what is where, and I also combined a bit the down path, because I believe it's relatively similar. For example:
- In both paths new issues (tickets) are being created
- Tickets, regardless of type, would have those four tags, be it documentation, or any other type, I presume
- In both cases the end is letting the VHD know

Author: Regarding the diagram, we had an extensive discussion with Mattias, and we don’t fully understand why, after a negative evaluation of a ticket, we need to proceed to the next step and assign tags.

Do we really need to tag something that has already been determined to be out of scope for the NHD? If so, I assume that the four different types of tags mentioned earlier in our SOP, such as "priority," would not apply to a ticket with a negative evaluation. Is this correct?

Author: ![image](https://github.com/user-attachments/assets/aa739ef1-377e-4d54-985b-78451b484422)
In your diagram, we thought to delete this connection after the negative evaluation with the Assign tags. 

Then the highlighted parts are not part of this SOP. Probably it would be a new "SOP with the ticket handling" that is going to include a lot of steps before the next step in your diagram "Update VHD on Ticket Resolution"

Reviewer: > ...why, after a negative evaluation of a ticket, we need to proceed to the next step and assign tags

Well, based on the G. document, it seemed that the process continues and you just tag it as "out of scope", or something similar. For example, the ticket status reads:

> Deleted - ticket that is archived e.g. since it is out of scope of the node HD

You could also add a new tag to the list, at a high level, that denotes "ticket out of scope".

In my mind, the process continues as with the positive evaluation, but it just gets resolved much faster:
- Okay this is out of scope
- I was the asignee
- The tag is "out of scope"
- The status is "delete"
- I tell the VHD

### 8. Procedure

#### 8.1. Evaluating a Ticket from the VHD

| Step identifier | When | Who |
|:---|:---|:---|

> | ``1`` | After receiving a ticket from the VHD, within xx working days | Helpdesk officer  |

Reviewer: Is the ``xx`` part up to the nodes?

Author: We were thinking along these lines: each node would decide within [XX working days] how many days they need to move forward to the next step. Ideally, it would be great if all nodes could follow the same process for handling tickets—for example, within 5 or 7 days. However, I’m not sure if this is feasible.

That said, considering the importance of consistency, it would be beneficial if all nodes agreed on the same timeframe for handling tickets.

Reviewer: I would recommend to add that for now to the SOP. If in the future we deem it impossible, or nodes want to modify it, then we'll discuss whether it's a compromise to break or not. But it's easier to do that than to add XX days in the future when we have given freedom from the beginning to set those timelines. And just like you say, it's something important for the Service Level Agreements.

Author: I understand, and there is a good point to what you say!  I changed it to 7 working days.

> | ``1`` | After receiving a ticket from the VHD, within xx working days | Helpdesk officer  |

Reviewer: Also, I would add that as a reminder below as well, within the text. Any deadline, try to highlight it in bold.

Author: We are a bit confused about the highlighting and the **bold format** .
In the example that you provided: 
![image](https://github.com/user-attachments/assets/0eb41fef-906e-43d6-a2d3-e96d5532a727)
It is bold the 2 months in the text after the table... Would you like us to do the same to the text after the table to highlight with bold the time period ... or this **To be done within xx working days**? Can it be bold inside the table?

Reviewer: It can be bold in both, but the bare minimum I would say is bold within the text, after the table. This is completely subjective though, but bold is useful to highlight these deadlines.

> The NHD receives a ticket from the VHD requiring action by the specific node. Using a node-specific checklist, the ticket is evaluated to determine whether it is within the scope (positive evaluation) of the node or not (negative evaluation).

Reviewer: I would format this part in a different way so that it's easier to read what the process is, and that this step is a decision making step.

- I would use **bold format**
- I would use bullet points, similar to what I had [here](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/sops/european-level/GDI-SOP0003_1%2Bmg-dac-recommendation-approval.md#82-recommend-application-for-approval). This way, you know that there are 2 options, and depending on the outcome of the "checking", the reader would know where to go (to step 8.2 or 8.3)
- I understand that the checklist is node-specific, so when nodes implement this SOP, they would add that part, right? If so, isn't it easy to end up using complete different criteria for whether a ticket is within the scope of the node? If possible I would add more context of what's required to be in that list, or an example of it, with redacted chunks to be filled in by the nodes later on (?)

Author: We agree that it would be nice to have a quite similar checklist for the different nodes, but we feel that it is not so easy for us to decide about the content of this checklist for evaluation if the ticket is or not in the scope of the GDI Node HD. We need more help from other helpdeskers.

Reviewer: I see. Even without the feedback of other nodes, I would change the format to be easier to follow if possible. Also, as mentioned, you can always format the scope part in a way it makes sense but with limited content.

For example, like:

> If the following criteria is true:
> - <node's scope criteria>
> - <node's scope criteria>
> 
> Continue to **step 2**.

Or something similar. Could be a phrase, or a checklist, or something, but in a way that different nodes say "Ah ok I'm supposed to add my node's criteria in these bullet points".

Author: I made some changes in this part, based on your example, which was a good suggestion. You can have a look!

#### 8.2.  Positive Evaluation - Proceeding with the Ticket**

| Step identifier | When | Who |
|:---|:---|:---|

> |  ``2`` | After step 1, if the evaluation is positive, within xx working days| Helpdesk officer |

Reviewer: I understand the "within xx working days", but I think it starts being confusing when there are multiple phrases and it's not step 1.

Maybe something like "**To be done** within xx..."?

Author: As I mentioned before, this to be done should replace the "within xx working days", but it should be bold in the table or in the text after the table. In other sops, it seems that nothing is bold in the tables apart from the headers.

Reviewer: It's alright if it's just bold after the table. If you want to highlight it within the table, that's alright as well. This part is up to us. The important bit is the rewording within the table, and the addition of it **after** the table

> If the ticket is within the scope of the NHD, a **new issue** will be created in the **dedicated system** to record the ticket's arrival and document the positive evaluation. Each ticket will be assigned **four tags** from the [provided list. ](https://docs.google.com/document/d/1KTAfpV5qLH9YTs8IXZWGXe_9InQg2ID3pAxWqoULrDU/edit?tab=t.0)

Reviewer: I formatted a bit the document, which made me think that you should probably create a named version of this file, one for reviewing and/or (depending on whether something changes) another for when it's submitted. I would also add the version of that document to this "provided list" thing. For example, "...from the provided list at ``List of tags for the SOP Node HD Ticket Classification - v1 21.02.2025](...)``".

This has to do with the fact that we're keeping hard versioning systems on the SOPs, but we are referencing something that is in the Google Drive and can easily be changed.

Author: We are quite confused about this Google document ([the provided list](https://docs.google.com/document/d/1KTAfpV5qLH9YTs8IXZWGXe_9InQg2ID3pAxWqoULrDU/edit?tab=t.0)). You wrote in your next message that it would be better to integrate it within this very SOP. What do you mean? 

To be part of the SOP procedure, this document? In this step, we mention the list step 8.2 to have all the content of the document, or should we refer to a document that we will save in another part of the GH? For example, in this part: 
![image](https://github.com/user-attachments/assets/5f36c39c-bf45-4ee0-b5cf-f1e7171939f4)

Reviewer: > You wrote in your next message that it would be better to integrate it within this very SOP

I mean that the textual list of the categories would be better when **embedded** within this SOP itself, rather than as a Google Doc referenced outside. This way, it's properly tracked.

We did not envision having a folder with documentation that SOPs refer to, because after all it's a style preference and little more. Documents that are way too long, and are maintained by other groups (e.g., the data privacy doc from Pillar I) can remain as a reference, but these documents (like this tag list) that we maintain and are referenced in the SOPs, would be better embedded.

I would simply add the Google document content as part of the SOP. If this step is "tagging", then add the way to tag and possible tags here too. If you want to separate ticket status, priority, etc. tagging in different steps, that's alright, and could be subdivided.

Reviewer: But I would recommend to have all that in the SOP itself, in this very markdown

> If the ticket is within the scope of the NHD, a **new issue** will be created in the **dedicated system** to record the ticket's arrival and document the positive evaluation. Each ticket will be assigned **four tags** from the [provided list. ](https://docs.google.com/document/d/1KTAfpV5qLH9YTs8IXZWGXe_9InQg2ID3pAxWqoULrDU/edit?tab=t.0)

Reviewer: Also, there's a bit of an issue in the wording, to my understanding: this text says "Each ticket till be assigned four tags **from the provided list**", but then:
- One of the tags is free text (the name of the assignee). I wouldn't personally consider a name to be a tag, besides.
- In the G. document itself, the wording is more sort of "this is a suggestion from EGA, but you can use other tags if you want". If that's the case, and each node is going to use theirs, then I would very much advocate for that document to be integrated within this very SOP. That way, not only the versioning is much easier to follow, but also when nodes do their implementations, they can change the used tags in this same document, and it would overall be easier than referencing different documents. 

So, as mentioned above, I would recommend to integrate within this step, the tags that are recommended, and mention that in nodes' implementations, one can modify the tags if needed for their system.

Author: Regarding the **assignee** as a tag, @mattiasstromberg believes that the **assignee** can be a tag. It is a big discussion this one. I do not know if you want to comment and explain further how you are thinking about tags in relation to assignee

Reviewer: In my opinion if you say "from the provided list", but then the list has open elements (like "add your name here"), then it's not a tag per se. Tags, to my knowledge, tend to be static, a selection of Controlled Vocabularies, that we as the creators select and curate, and we provide as a list to choose from to the ones tagging.

Even without the "asignee" "tag", the wording should be addressed, so that it's not contradictory between "These tags are the only ones you can choose from" and "these are just suggestions".

Reviewer: IMHO it would be better to have the asignee as simply a custom text, not call it a tag, so that the wording is easier for tags. And also that we enforce tags, not "just a suggestion", but rather "this are the tags. Choose". And then, in the future, if needed other nodes could edit that list of tags **within this document** in their implementations. But originally, at least, we got the list. 

At most, if we're going to give free will to change these tags, I would add a comment or a phrase somewhere that says that these tags may change node to node.

> **Tags**: Assign these four tags to each ticket:

Reviewer: If the diagram that I suggested, which also entails a different step workflow, is deemed worth it, this tag assignment would be a different step in itself.

Author: If we understood correctly, you suggest that we will create a new step in the following diagram: 
![image](https://github.com/user-attachments/assets/ac13780d-6e86-4fff-a9cd-f7eb005d46a4)

After the positive evaluation we create a new issue (one step) and assign tags (second step) ...

Reviewer: I think I meant it more as in create a step for each of the boxes (at least) in the diagram. That way, you can phrase each step generically, and it's easier to reuse steps.

For example, if I'm adding tags, it's not "positive ticket - time to add tags", but simply "add tags", and then within the step say:
- If it's positive evaluation, check these tags out
- If it's negative evaluation, add the tag "out of scope"

After all, these suggestions about which tags to use are just that, suggestions, because it's supposed to be more generic, like I say, as in "And now you add the tags". But we obviously want to help people, rather than letting them go through 100 tags, where one is "out of scope".

Author: I have some further questions about the diagram ... is it supposed to create an **issue** (if positive evaluation) and to create a **"documentation issue"** (negative evaluation), are these two different things? And we need two different boxes? 

Then I do not understand why we need to **inform VHD of out-of-scope decision** and then to have the other **update VHD on ticket resolution**. Why should we have two steps for that? For the negative evaluation, it might be enough with one email or message to inform them about the out-of scope decision and the ticket resolution.  From the diagram I get a feeling that these are two different steps that I must do for the negative evaluation. 

This **follow NHD SOP for ticket handling** is something that is associated with tickets with negative evaluation? My interpretation is that SOP will be only for the tickets with positive evaluation only ... 

![image](https://github.com/user-attachments/assets/1c459a08-2d47-4ea5-aa9b-99fd1d7a10e9)

Reviewer: > ...why we need to inform VHD of out-of-scope decision and then to have the other update VHD on ticket resolution

We may very well not need the first (``VHD of out-of-scope decision``), indeed. We just need to make it clear that the resolution of a negatively evaluated ticket will be communicated in the same way that the resolution of any other ticket in the NHD would.

All in all, it's to make the steps simpler, by making them reusable, regardless of the initial evaluation.

> My interpretation is that SOP will be only for the tickets with positive evaluation only

I would expect for a very quick/short SOP on how to solve out of scope issues would exist.

In simpler words, I would see the following:
1. VHD gets a request
2. NHD gets the ticket from VHD
3. NHD checks if the ticket is out of scope or not.
4. NHD Creates an internal ticket. 
   1. This ticket can be of different types. One of which is "out of scope", or "documentation", or anything else that clearly depicts "hey this was given by the VHD, but it's not in our scope. Nevertheless, we created the issue internally in our NHD so that it's traceable".
5. The internal ticket at NHD, **regardless of the type**, continues the flow, and gets tags, assignee, etc.
6. The internal ticket at NHD, **regardless of the type**, continues the flow, and gets resolved as needed.
   1. Each ticket type may have its own SOP, or maybe all are aggregated in one big SOP. Regardless, both positive and negative evaluations follow the same path: they get resolved.
   2. The way to resolve them is what changes. In some cases it'll be a simple "Hey I'm closing this ticket, because as the tag says it's out of scope. Done". In some other cases, it'll take days and weeks doing more complicated stuff, etc.
7. The NHD tells the VHD that the internal ticket, corresponding to their initial issue, was completed. This applies for both the out of scope and in scope. It's simply that the message reads differently, but it's essentially the same.

Please note that the numbering in this oversimplification does not necessarily need to correspond to the steps at all.

Author: I have updated the diagram and incorporated it into the SOP text based on your suggestions, which I believe are quite good. Please take a look at my updates.

My only question concerns the very last step: should we explicitly include somewhere that the ticket is 'closed'? While it is resolved, our current ticket statuses do not include a 'Closed' status. Do we need to add this, and if so, where in the diagram should it be placed?

![image](https://github.com/user-attachments/assets/1ad54a5c-bea6-4615-87f8-5eb585ecace3)

![image](https://github.com/user-attachments/assets/0789c8a3-e85c-457b-8b13-1cd9509cbbf3)

Reviewer: That's good, I see it logically.
Re. the closing ticket, I don't know myself, but I was expecting for it to be part of the "Follow NHD SOP for ticket handling". As in, it would be how to resolve the ticket and handle it until the very end when it's closed and you notify the VHD.

1. **Type**: Specifies the type of ticket.

2. **Priority**: Specifies the ticket's priority level.

3. **Assignee**: Specifies who is responsible for the ticket.

4. **Status**: Specifies the ticket's status (initially set to "New").

#### 8.3.  Negative Evaluation - Closing the Ticket

| Step identifier | When | Who |
|:---|:---|:---|
| ``3`` | After step 1, if the evaluation is negative, within xx working days | Helpdesk officer |

If the ticket is out of the scope of the NHD, a new issue will be created in the designated system to record the ticket's arrival and document the negative evaluation.

##### 8.3.1 Create a new issue in the system
| Step identifier | When | Who |
|:---|:---|:---|
| ``3.1`` | After creating a new issue in the dedicated system and documenting the negative evaluation | Helpdesk officer |

> The VHD will be informed that the issue is not within the scope of the NHD. A brief explanation for this decision will be provided.

Reviewer: If available, I would add here a template of the message, and the method of delivery. For example, to which mailing list (if by email) the NHD tells the VHD that it's not in scope. 

Also the template of the message, similar to **[this](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/sops/european-level/GDI-SOP0007_sop-template-creation.md#853-contact-management-board)**. It would make it easier to fetch and report when those communications happened, how often, from whom, etc.

Author: I added an email template to the SOP.

> ##### 8.3.2 Inform the VHD and close the ticket

Reviewer: I think these steps titles are mixed. The VHD was informed in the previous one I understand (?)

Reviewer: Also note the step flow of the diagram proposed above, which I think is more granular and slightly easier to follow. Subjective, though!

Author: You are right about the information on VHD. :-)  We are going to correct this part. Do you mean we must include these two steps from the diagram in the procedure part of the SOP?
![image](https://github.com/user-attachments/assets/fc9d4c4b-0c1a-4ed4-b16f-d2dfaf12fe02)

Reviewer: I think that, if you agree with the diagram steps, it's best to have a step for each of the diagram steps. As mentioned above.

Reviewer: It's subjective though

Author: I have done this ... I tried to create a step based on the diagram steps. Based on the new diagram, though :-)

| Step identifier | When | Who |
|:---|:---|:---|
| ``3.2`` | After informing the VHD about the negative evaluation of the issue  | Helpdesk officer |

The ticket will be closed in the node’s ticketing system, with a clear note stating why the ticket was closed.

> ### 9. References

Reviewer: We would need to have in the references the Google Doc that was referenced above, unless of course we embed it inside this SOP itself, which would be my advise.

Author: We will try to embed this document inside the SOP, but we need to clarify in which part of the GH repository we will include the list (in the SOP procedures? or in a document in the GH that we will refer to )

Reviewer: It's a bit up to us. If you guys feel that the list is way too long, we can see whether to create another SOP or not. Otherwise, within the doc would be ok. After all, we can reference things in the doc with headers, and it's not like markdown has a limit. It would just mean more scrolling, or simply clicking on the index.

If embedded, to ease the reading, I would advise to add a "back to top of tag list" hyperlink, where if you click, it brings you up to the top of that step again. Or something like that, just to ease the traversing of steps.
